compiler/ir/backend.wasm/src/org/jetbrains/kotlin/backend/wasm/ir2wasm/WasmFunctionCodegenContextImpl.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.backend.wasm.ir2wasm
+
+import org.jetbrains.kotlin.backend.wasm.WasmBackendContext
+import org.jetbrains.kotlin.ir.declarations.IrFunction
+import org.jetbrains.kotlin.ir.declarations.IrValueParameter
+import org.jetbrains.kotlin.ir.expressions.IrLoop
+import org.jetbrains.kotlin.ir.symbols.IrValueParameterSymbol
+import org.jetbrains.kotlin.ir.symbols.IrValueSymbol
+import org.jetbrains.kotlin.wasm.ir.*
+
+class WasmFunctionCodegenContextImpl(
+    override val irFunction: IrFunction,
+    private val wasmFunction: WasmFunction.Defined,
+    override val backendContext: WasmBackendContext,
+    private val referencing: WasmBaseCodegenContext
+) : WasmBaseCodegenContext by referencing,
+    WasmFunctionCodegenContext {
+    override val bodyGen: WasmExpressionBuilder =
+        WasmIrExpressionBuilder(wasmFunction.instructions)
+
+    private var labelIds = 0
+    override fun getNextLabelId(): Int = labelIds++
+
+    private val wasmLocals = LinkedHashMap<IrValueSymbol, WasmLocal>()
+    private val loopLevels = LinkedHashMap<Pair<IrLoop, LoopLabelType>, Int>()
+
+    private var localIds: Int = 0
```

Comment:
read locals count from wasmFunction.locals or ensure it's 0

Reply:
Replaced with wasmFunction.locals.size